pkgs/tools/virtualization/xe-guest-utilities/default.nix

```
@@ -1,82 +1,39 @@
-{ lib, stdenv, fetchurl, bzip2, lzo, zlib, xz, bash, python2, gnutar, gnused, gnugrep, which }:
+{ lib
+, buildGoModule
+, fetchFromGitHub
+}:
 
-stdenv.mkDerivation (rec {
+buildGoModule rec {
   pname = "xe-guest-utilities";
-  version = "6.2.0";
-  meta = {
-    description = "Citrix XenServer Tools";
-    homepage = "http://citrix.com/English/ps2/products/product.asp?contentID=683148&ntref=hp_nav_US";
-    maintainers = with lib.maintainers; [ benwbooth ];
-    platforms = lib.platforms.linux;
-    license = [ lib.licenses.gpl2 lib.licenses.lgpl21 ];
-    # never built on aarch64-linux since first introduction in nixpkgs
-    broken = stdenv.isLinux && stdenv.isAarch64;
-  };
-  src = fetchurl {
-    url = "https://sources.archlinux.org/other/community/xe-guest-utilities/xe-guest-utilities_${version}-1120.tar.gz";
-    sha256 = "f9593cd9588188f80253e736f48d8dd94c5b517abb18316085f86acffab48794";
+  version = "7.30.0";
+
+  src = fetchFromGitHub {
+    owner = "xenserver";
+    repo = "xe-guest-utilities";
+    rev = "v${version}";
+    hash = "sha256-gMb8QIUg8t0SiTtUzqeh4XM5hHgCXuf5KlV3OeoU0LI=";
   };
-  buildInputs = [ bzip2 gnutar gnused python2 lzo zlib xz gnugrep which ];
-  patches = [ ./ip-address.patch ];
+
+  vendorHash = "sha256-zhpDvo8iujE426/gxJY+Pqfv99vLNKHqyMQbbXIKodY=";
+
   postPatch = ''
-    tar xf "$NIX_BUILD_TOP/$name/xenstore-sources.tar.bz2"
+    substituteInPlace Makefile \
+      --replace 'RELEASE := $(shell git rev-list HEAD | wc -l)' ""
   '';
 
-  # Workaround build failure on -fno-common toolchains:
-  #   ld: utils.o:xenstore/utils.h:27:
-  #     multiple definition of `xprintf'; xenstored_core.o:xenstore/utils.h:27: first defined here
-  NIX_CFLAGS_COMPILE = "-fcommon";
-
   buildPhase = ''
-    export CC=gcc
-    export CFLAGS='-Wall -Wstrict-prototypes -Wno-unused-local-typedefs -Wno-sizeof-pointer-memaccess'
-    export PYTHON=python2
-    cd "$NIX_BUILD_TOP/$name/uclibc-sources"
-    for file in Config.mk tools/libxc/Makefile tools/misc/Makefile tools/misc/lomount/Makefile tools/xenstore/Makefile; do
-      substituteInPlace "$file" --replace -Werror ""
-    done
-    make -C tools/include
-    make -C tools/libxc
-    make -C tools/xenstore
+    make
```

And we should be able to remove `postPatch`

```suggestion
    make RELEASE=nixpkgs
```